src/test/scala/re2s/ExecTest.scala

```diff
@@ -0,0 +1,686 @@
+
+import org.scalatest.FunSuite
+
+// TestRE2 tests this package's regexp API against test cases
+// considered during (C++) RE2's exhaustive tests, which run all possible
+// regexps over a given set of atoms and operators, up to a given
+// complexity, over all possible strings over a given alphabet,
+// up to a given size.  Rather than try to link with RE2, we read a
+// log file containing the test cases and the expected matches.
+// The log file, re2.txt, is generated by running 'make exhaustive-log'
+// in the open source RE2 distribution.  http://code.google.com/p/re2/
+//
+// The test file format is a sequence of stanzas like:
+//
+//      strings
+//      "abc"
+//      "123x"
+//      regexps
+//      "[a-z]+"
+//      0-3;0-3
+//      -;-
+//      "([0-9])([0-9])([0-9])"
+//      -;-
+//      -;0-3 0-1 1-2 2-3
+//
+// The stanza begins by defining a set of strings, quoted
+// using Go double-quote syntax, one per line.  Then the
+// regexps section gives a sequence of regexps to run on
+// the strings.  In the block that follows a regexp, each line
+// gives the semicolon-separated match results of running
+// the regexp on the corresponding string.
+// Each match result is either a single -, meaning no match, or a
+// space-separated sequence of pairs giving the match and
+// submatch indices.  An unmatched subexpression formats
+// its pair as a single - (not illustrated above).  For now
+// each regexp run produces two match results, one for a
+// ``full match'' that restricts the regexp to matching the entire
+// string or nothing, and one for a ``partial match'' that gives
+// the leftmost first match found in the string.
+//
+// Lines beginning with # are comments.  Lines beginning with
+// a capital letter are test names printed during RE2's test suite
+// and are echoed into t but otherwise ignored.
+//
+// At time of writing, re2.txt is 32 MB but compresses to 760 kB,
+// so we store re2.txt.gz in the repository and decompress it on the fly.
+class ExecTest() extends FunSuite {
+  test("ExamplesInDocumentation") {
+    val re = RE2.compile("(?i:co(.)a)")
+    assert(
+      Array("Copa", "coba").deep == re.findAll("Copacobana", 10).toArray.deep)
+    val x = re.findAllSubmatch("Copacobana", 100)
+    assert(Array("Copa", "p").deep == x.get(0).deep)
+    assert(Array("coba", "b").deep == x.get(1).deep)
+  }
+
+  test("RE2Search") { // 1832pass/20failed
+    testRE2("re2-search.txt")
+  }
+
+  test("RE2Exhaustive") { // >100failed
+    testRE2("re2-exhaustive.txt.gz") // takes about 30s
+  }
+
+  def testRE2(_file: String) = {
+    var file = _file
+    var in   = classOf[ExecTest].getResourceAsStream("/" + file)
+    // TODO(adonovan): call in.close() on all paths.
+    if (file.endsWith(".gz")) {
+      in = new GZIPInputStream(in)
+      file = file.substring(0, file.length - ".gz".length) // for errors
+    }
+    var lineno  = 0
+    val r       = new UNIXBufferedReader(new InputStreamReader(in, "UTF-8"))
```

Comment:
`scala.io.Source.fromResource(resource: String,classLoader: ClassLoader)`

Reply:
I'd not recommend using `scala.io`. It's an old API that's likely to be on its way out (nobody is working on it, and new alternatives like betterfiles don't depend on it).

Reply:
Make sense!